packages/expo-ui/components/ListItem/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { requireNativeView } from 'expo';
+import { Children, Fragment, isValidElement, ReactElement, ReactNode, useMemo } from 'react';
+import { NativeSyntheticEvent } from 'react-native';
+
+import { Button, ButtonProps, NativeButtonProps, transformButtonProps } from '../Button';
+
+type SwipeActionNativeEvent = NativeSyntheticEvent<{ id: string }>;
+type EventHandlerMap = Map<string, (event: SwipeActionNativeEvent) => void>;
+type NativeSwipeActionProps = { id: string; props: NativeButtonProps };
+type SwipeActionElement = ReactElement<ButtonProps>;
+
+type SwipeAction = SwipeActionElement | React.ReactElement<{ children: SwipeActionElement[] }>;
+
+export type ListItemProps = {
+  /**
+   * Actions which will appear when swiping from the leading edge
+   */
+  leadingActions?: SwipeAction;
+
+  /**
+   * Whether to activate the first action when user overshoots swipe from leading edge
+   * @default true
+   */
+  allowsFullSwipeLeading?: boolean;
+
+  /**
+   * Actions which will appear when swiping from the trailing edge
+   */
+  trailingActions?: SwipeAction;
+
+  /**
+   * Whether to activate the first action when user overshoots swipe from trailing edge
+   * @default true
+   */
+  allowsFullSwipeTrailing?: boolean;
+
+  /**
+   * Whether to hide trailing edge actions when list edit mode is active
+   * @default true
+   */
+  hideTrailingActionsInEditMode?: boolean;
```

Comment:
Swipe actions overtake the parent list’s delete action by default. But it’s kind of confusing, so I opted for disabling it by default. You can enable it using this prop

https://github.com/user-attachments/assets/ee51d795-49ea-4f7c-a19e-ea4fa3ae1b02